neqo-crypto/tests/agent.rs

```diff
@@ -535,3 +536,100 @@ fn ech_retry() {
     assert!(client.preinfo().unwrap().ech_accepted().unwrap());
     assert!(server.preinfo().unwrap().ech_accepted().unwrap());
 }
+
+#[test]
+fn connection_succeeds_when_server_and_client_support_cert_compr_copy() {
+    struct CopyCompression {}
+
+    // Implementation supports both encoder and decoder
+    impl SafeCertificateCompression for CopyCompression {
+        const ID: u16 = 0x4;
+        const NAME: &str = concat!("copy", "\0");
+        const ENABLE_ENCODING: bool = true;
+
+        fn encode(data: &[u8]) -> Vec<u8> {
+            data.to_vec()
+        }
+        fn decode(data: &[u8]) -> Vec<u8> {
+            data.to_vec()
+        }
+    }
+
+    fixture_init();
+    let mut client = Client::new("server.example", true).expect("should create client");
+    let mut server = Server::new(&["key"]).expect("should create server");
+
+    client
+        .set_certificate_compression::<CopyCompression>()
+        .unwrap();
+    server
+        .set_certificate_compression::<CopyCompression>()
+        .unwrap();
+
+    connect(&mut client, &mut server);
+
+    assert!(client.state().is_connected());
+    assert!(server.state().is_connected());
```

Comment:
These functions need to test that compression is active as well.  That means checking on whatever information about compression NSS has.

Reply:
I believe that NSS does not know anything about the compression we use. 

What we can do is to extend [SSLChannelInfoStr ](https://searchfox.org/mozilla-central/source/security/nss/lib/ssl/sslt.h#305) with the information of whether/which encoder is used. What do you think? P.s. `SSLCompressionMethod compressionMethod;` not to be mistaken for rfc8879 certificate compression

Reply:
Yeah, I'm happy to defer that to a second change here, but I really do think that we need to be able to report a) whether compression was used, and b) what the compression saved us.  That probably means extending SSLChannelInfo with new fields.

Reply:
Do you mind if we land this patch first and then extend the SSLChannelInfo in the following patch?

Reply:
Absolutely not.  I'd prefer it if we did it that way.  You'll need time to work out the NSS patch and I don't want this patch sitting here rotting in the meantime.